ts/main.ts

```diff
@@ -29,26 +34,50 @@ $urlLinkInput.addEventListener('input', (event: Event) => {
 $entryForm.addEventListener('submit', (event: Event) => {
   event.preventDefault();
   const $formElements = $entryForm.elements as FormElements;
-  const $formObject: FormObject = {
-    title: $formElements.title.value,
-    url: $formElements.url.value,
-    notes: $formElements.notes.value,
-    entryID: data.nextEntryId,
-  };
-  data.nextEntryId++;
-  data.entries.unshift($formObject);
-  const $newEntry = renderEntry($formObject);
-  $ul?.prepend($newEntry);
-  viewSwap('entries');
-  if (data.entries.length > 0) {
-    toggleNoEntries();
+  if (data.editing === null) {
+    const $formObject: FormObject = {
+      title: $formElements.title.value,
+      url: $formElements.url.value,
+      notes: $formElements.notes.value,
+      entryID: data.nextEntryId,
+    };
+    data.nextEntryId++;
+    data.entries.unshift($formObject);
+    const $newEntry = renderEntry($formObject);
+    $ul.prepend($newEntry);
+    if (data.entries.length > 0) {
+      toggleNoEntries();
+    }
+  } else {
+    const $formObject: FormObject = {
+      title: $formElements.title.value,
+      url: $formElements.url.value,
+      notes: $formElements.notes.value,
+      entryID: data.editing.entryID,
+    };
+    const updatingIndex = data.entries.findIndex(
+      (entry: FormObject) => entry.entryID === $formObject.entryID
+    );
+    data.entries[updatingIndex] = $formObject;
+    $ul.textContent = '';
+    data.entries.forEach((entry: FormObject) => {
+      const $newEntry = renderEntry(entry);
+      $ul.append($newEntry);
+      $urlLinkInput.removeAttribute('value');
+      $titleInput?.removeAttribute('value');
+      $notesInput.textContent = '';
+      $entryTitle.textContent = 'New Entry';
+      data.editing = null;
+    });
```

Comment:
Rather than removing all of the children and recreating them, you should find the one that already exists and just replace it. We don't want to regenerate the children every time